src/nodes/victron-virtual.js

```diff
@@ -804,231 +802,229 @@ module.exports = function (RED) {
+            }
 
-              const groupKey = `SwitchableOutput/output_${i}/Settings/Group`
-              ifaceDesc.properties[groupKey] = {
-                type: 's',
+            // Temperature setpoint
+            if (switchType === 3) {
+              // Dimming value (°C)
+              const dimmingKey = 'SwitchableOutput/output_1/Dimming'
+              ifaceDesc.properties[dimmingKey] = {
+                type: 'd',
+                format: (v) => v != null ? v.toFixed(1) + '°C' : '',
+                min: Number(config.switch_1_min ?? 0),
+                max: Number(config.switch_1_max ?? 100),
                 persist: true
               }
-              iface[groupKey] = config[`switch_${i}_group`] || ''
+              iface[dimmingKey] = Number(config.switch_1_initial ?? 0)
 
-              if (switchType === 2) {
-                const dimmingKey = `SwitchableOutput/output_${i}/Dimming`
-                ifaceDesc.properties[dimmingKey] = {
-                  type: 'd',
-                  format: (v) => v != null ? v.toFixed(1) + '%' : '',
-                  min: 0,
-                  max: 100,
-                  persist: true
-                }
-                iface[dimmingKey] = 0
+              // DimmingMin
+              const minKey = 'SwitchableOutput/output_1/Settings/DimmingMin'
+              ifaceDesc.properties[minKey] = {
+                type: 'd',
+                format: (v) => v != null ? v.toFixed(1) + '°C' : ''
               }
+              iface[minKey] = Number(config.switch_1_min ?? 0)
 
-              // Temperature setpoint
-              if (switchType === 3) {
-                // Dimming value (°C)
-                const dimmingKey = `SwitchableOutput/output_${i}/Dimming`
-                ifaceDesc.properties[dimmingKey] = {
-                  type: 'd',
-                  format: (v) => v != null ? v.toFixed(1) + '°C' : '',
-                  min: Number(config[`switch_${i}_min`] ?? 0),
-                  max: Number(config[`switch_${i}_max`] ?? 100),
-                  persist: true
-                }
-                iface[dimmingKey] = Number(config[`switch_${i}_initial`] ?? 0)
+              // DimmingMax
+              const maxKey = 'SwitchableOutput/output_1/Settings/DimmingMax'
+              ifaceDesc.properties[maxKey] = {
+                type: 'd',
+                format: (v) => v != null ? v.toFixed(1) + '°C' : ''
+              }
+              iface[maxKey] = Number(config.switch_1_max ?? 100)
 
-                // DimmingMin
-                const minKey = `SwitchableOutput/output_${i}/Settings/DimmingMin`
-                ifaceDesc.properties[minKey] = {
-                  type: 'd',
-                  format: (v) => v != null ? v.toFixed(1) + '°C' : ''
-                }
-                iface[minKey] = Number(config[`switch_${i}_min`] ?? 0)
+              // StepSize
+              const stepKey = 'SwitchableOutput/output_1/Settings/StepSize'
```

Comment:
There is an issue with the "source of truth" of the StepSize: The StepSize is stored both in the configuration (in the flow) and in dbus, and both values are writable, and they are not getting synced from dbus to the configuration.

The UI uses the step size available via dbus.

This means I can change the step size via dbus, which will reflect in the UI. However, when I deploy the flow again, the step size will be overwritten from the flow.

We cannot easily sync from dbus to the flow, so that is not a solution.

When opening the configuration, can we read from dbus at that moment? This may also not be easy.

I'm not sure what a good solution is.

This issue may exist for other configuration properties as well, I've been testing and investigating using the step size.

Reply:
Probably the best way to solve this is to add a disclaimer on the edit panel. E.g. with something like:
> Note that if you change these settings via a control node, the configuration value shown here might be out of sync.

As this is only relevant for users that actually do change these values, they are probably also fine with accepting that the reasoning for the values getting out of sync.